### PR TITLE
Docs add

### PR DESCRIPTION
**Pull request by poiresel**
## Description

Quick description:

- [X] Documentation

To use; if you're looking to download a solcjson.js file look here to target specific versions. Please go here: https://github.com/ethereum/solc-bin
I have a todo in the repo solidity-docgen to make it just work with someone putting in a solidity version like 0.5.12
## Tests

N/A
## Test Coverage
N/A

### CODE REVIEW

File: source/delegate-factory/contracts/DelegateFactory.sol [hunk 1]
```diff
@@ -67,7 +73,7 @@ contract DelegateFactory is IDelegateFactory, ILocatorWhitelist {
   }
 
   /**
-    * @notice To check whether a locator was deployed
+    * @notice To check whether a locator was deployed from this factory
```
Comment (ejwessel) [inline source/delegate-factory/contracts/DelegateFactory.sol]: `To check whether` -> `Checks whether`

File: source/delegate-factory/docsTemplate/contract.hbs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+{{#if natspec.returns}}
+|  Param    | Description   |
+{{#each natspec.returns}}
+{{#if @first}}
+|   ---------| :------------ |
+{{/if}}| `{{param}}`|  {{description~}}{{#raw}}|{{/raw}}{{/each}}{{/if}}
+
+{{/ownFunctions}}
+
+## Price Calculations
```
Comment (ejwessel) [inline source/delegate-factory/docsTemplate/contract.hbs]: is there a reason we need price calculations in the delegate-factory docs when they're already in the delegate docs?

File: source/types/docsTemplate/contract.hbs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+{{#if natspec.returns}}
+|  Param    | Description   |
+{{#each natspec.returns}}
+{{#if @first}}
+|   ---------| :------------ |
+{{/if}}| `{{param}}`|  {{description~}}{{#raw}}|{{/raw}}{{/each}}{{/if}}
+
+{{/ownFunctions}}
+
+## Price Calculations
```
Comment (ejwessel) [inline source/types/docsTemplate/contract.hbs]: same question on price calc here if we have it in delegate.

File: source/delegate-factory/docsTemplate/contract.hbs (hunk 2 above)
Comment (poiresel) [reply source/delegate-factory/docsTemplate/contract.hbs]: will remove.

File: source/types/docsTemplate/contract.hbs (hunk 3 above)
Comment (poiresel) [reply source/types/docsTemplate/contract.hbs]: will remove.

File: source/swap/docs/Swap.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+Perform an ERC-20 or ERC-721 token transfer
+
+
+Transfer type specified by the bytes4 kind param
+ERC721: uses transferFrom for transfer
+ERC20: Takes into account non-standard ERC-20 tokens.
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`from`|address Wallet address to transfer from
```
Comment (alicevhenshaw) [inline source/swap/docs/Swap.md]: Is there a way to split the type out into its own column?

File: source/types/docsTemplate/contract.hbs [hunk 5]
```diff
@@ -0,0 +1,103 @@
+## <span id="{{anchor}}"></span> `{{name}}`
+
+ Delegate is a smart contract that sends orders based on rules. View the code on GitHub.
```
Comment (alicevhenshaw) [inline source/types/docsTemplate/contract.hbs]: This is the types doc. This line is for the delegate

File: source/wrapper/docsTemplate/contract.hbs [hunk 6]
```diff
@@ -0,0 +1,69 @@
+## <span id="{{anchor}}"></span> `{{name}}`
+
+`{{{natspec.title}}}`
+
+Wrapper is a shim over the Swap contract. The Swap contract only supports tokens, so for ether (ETH) to be used it must first be wrapped into WETH. Orders can be submitted to the Wrapper, which deals with the wrapping and unwrapping of ETH, as well as passing the Order onwards to the Swap contract. Swaps through the wrapper cannot be performed with authorized senders or blank signatures (See Swap for more detail on these cases). View the code on GitHub.
```
Comment (alicevhenshaw) [inline source/wrapper/docsTemplate/contract.hbs]: In Delegate the explanation goes above `natspec.title`, and here it goes below

File: source/indexer/docsTemplate/contract.hbs [hunk 7]
```diff
@@ -0,0 +1,55 @@
+## <span id="{{anchor}}"></span> `{{name}}`
+
+`{{{natspec.title}}}`
+
```
Comment (alicevhenshaw) [inline source/indexer/docsTemplate/contract.hbs]: No summary of the contract provided

File: source/swap/docs/Swap.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizer`|address Address doing the authorization
+|`delegate`|address Address being authorized
+
+
+|  Param    | Description   |
+|   ---------| :------------ |
+| `bool`|  True if a delegate is authorized to send
```
Comment (alicevhenshaw) [inline source/swap/docs/Swap.md]: Can we add a title or a line to make it clear that this is the return values table? It looks the exact same as the parameters

File: source/swap/docs/Swap.md (hunk 4 above)
Comment (poiresel) [reply source/swap/docs/Swap.md]: next iteration, working on modifying solidity-docgen.

File: source/swap/docsTemplate/contract.hbs [hunk 9]
```diff
@@ -34,7 +34,9 @@
 | ----------------- | :------------    |
 {{/if}}|`{{param}}`|{{description~}}{{#raw}}|{{/raw}}{{/each}}{{/if}}
 
+
 {{#if natspec.returns}}
+## Returns
```
Comment (alicevhenshaw) [inline source/swap/docsTemplate/contract.hbs]: I think you only added the `## returns` line to one of the template docs. Please can you add it to all of them?

File: source/swap/docs/Imports.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
+## <span id="Imports"></span> `Imports`
+
+``
```
Comment (alicevhenshaw) [inline source/swap/docs/Imports.md]: What are these for?

Comment (alicevhenshaw) [reply source/swap/docs/Imports.md]: Actually just confused by this whole doc. Why does Imports need a readme that explains authorizations?

File: source/swap/docs/Migrations.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+### <span id="Migrations-setCompleted-uint256-"></span> `setCompleted(uint256 completed)` (public)
+
+
+
+
+
+
+
+
+### <span id="Migrations-upgrade-address-"></span> `upgrade(address newAddress)` (public)
```
Comment (alicevhenshaw) [inline source/swap/docs/Migrations.md]: Same with migrations - do we need it?

File: source/swap/docs/Swap.md [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+
+### <span id="Swap-isSenderAuthorized-address-address-"></span> `isSenderAuthorized(address authorizer, address delegate) → bool` (internal)
+
+Determine whether a sender delegate is authorized
+
+
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizer`|address Address doing the authorization
```
Comment (alicevhenshaw) [inline source/swap/docs/Swap.md]: Do these need a final | on the rows?

File: source/swap/docs/Swap.md [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+Determine whether a signer delegate is authorized
+
+
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizer`|address Address doing the authorization
+|`delegate`|address Address being authorized
+
+
```
Comment (alicevhenshaw) [inline source/swap/docs/Swap.md]: this still doesnt have a ## returns

**Pull request closed without merging**